File: Javascript/Array/M-151. Reverse Words in a String.ts
function reverseWords(s: string): string {
  const arr = s.split(' ').filter((char) => char !== '');
  const reverse = arr.reverse().join(' ');
  return reverse;
}

const testData = [
  {
    s: 'the sky is blue',
    ans: 'blue is sky the',
  },
  {
    s: '  hello world  ',
    ans: 'world hello',
  },
  {
    s: 'a good   example',
    ans: 'example good a',
  },
];

for (const data of testData) {
  console.log(data['ans'] === reverseWords(data['s']));
}

File: Array/M-151. Reverse Words in a String/M-151. Reverse Words in a String.py
class Solution:
    @classmethod
    def reverseWords(self, s: str) -> str:
        arr = list(filter(lambda char: char != "", s.split(" ")))
        arr.reverse()
        reversed_str = " ".join(arr)

        return reversed_str


testData = [
    {
        "s": "the sky is blue",
        "ans": "blue is sky the",
    },
    {
        "s": "  hello world  ",
        "ans": "world hello",
    },
    {
        "s": "a good   example",
        "ans": "example good a",
    },
]

for data in testData:
    print(data["ans"] == Solution.reverseWords(data["s"]))
